### ISSUE TITLE
About node feature in the guidance graph

### ISSUE THREAD

**Issue by yuxu-1** [gao-lab/GLUE#91]
Hi,

I read your paper and did not find the text about initialized node features in the guidance graph (maybe I missed the relevant text..). 

In my understanding, each node in that graph is a gene (atac-seq peak region, or methylation region), which is irrelevant to cells. I mean, these nodes can be defined without single-cell data. Is that right?

And I found the initialization of node features in `sc.py` -> `class GraphEncoder(glue.GraphEncoder)` -> line 38 `self.vrepr = torch.nn.Parameter(torch.zeros(vnum, out_features))`. So the nodes in the guidance graph have no context information, and the graph is built by edge relationships (atac peak/methylation in the promoter of a gene) only. I am not sure whether my understanding is right or not.

Thanks
Yu

**Comment by Jeff1995** [comment]
Hi Yu! Thanks for your interest in GLUE. You are right about the identity of the nodes, but the graph encoder in GLUE does not use node features, the input of graph convolution is learnable node embeddings ([`vrepr`](https://github.com/gao-lab/GLUE/blob/e006de995b0b615d679287e620eaa49e2d8afb3f/scglue/models/sc.py#L38)).

You may also interpret it as one-hot node features times a linear transformation $I_{\mathrm{vnum} \times \mathrm{vnum}} \cdot V_{\mathrm{vnum} \times \mathrm{out \textunderscore features}}$, where the linear transformation is exactly the node embeddings.

Hope that clarifies. Please let me know if there were further questions!

**Comment by yuxu-1** [comment]
Hi Jeff, 

Thanks for your reply. GLUE is a great research work!
I still have a question: the learnable node embeddings (vrepr,  $V_{vnum\times out_features}$) were initialized with the scalar value 0, which would make the [message](https://github.com/gao-lab/GLUE/blob/e006de995b0b615d679287e620eaa49e2d8afb3f/scglue/models/nn.py#L50) in the process of GCN to be 0. How do you transmit information in the graph?


BR
Yu

**Comment by Jeff1995** [comment]
While the embeddings are initialized as zeros, they will be updated during training as guided by the graph and data decoders. They will definitely be non-zero after the model has been trained.

**Comment by yuxu-1** [comment]
Hi Jeff,

I see. Thanks! 
The learnable node embeddings (torch parameters) have gradients after backforward.


BR
Yu
